dask_ml/model_selection/_search.py

```diff
@@ -224,6 +232,20 @@ def build_cv_graph(
     cv_name = "cv-split-" + main_token
     dsk[cv_name] = (cv_split, cv, X_name, y_name, groups_name, is_pairwise, cache_cv)
 
+    eval_weight_source = _get_weights_source(fit_params)
+    if eval_weight_source is not None :
+        weights = "cv-n-weights-" + main_token
```

Comment:
The code in this block is replicated exactly above at lines 160 to 165. Would it make sense to DRY this out and put it in a function?

Reply:
put into new function